index.rst

```diff
@@ -1,61 +1,311 @@
+In order to allow quick turnaround, the command can accept an additional yaml string with overriding  parameters for the currently loaded configuration.
+
+For instance, one could truncate the sensitivity matrix so that a wavefront error correction is applied to a specific component (or components), e.g.;
+
+.. code-block:: py
+
+  import yaml
+
+  # switch off all corrections except m2 hexapod
+  # See https://github.com/lsst-ts/ts_ofc/blob/master/policy/zkAndDofIdxArraySet.yaml
+  ofc_config = {
+    "dofIdx": {
+        "m2Hex": [1, 1, 1, 1, 1],
+        "camHex": [0, 0, 0, 0, 0],
+        "m1m3Bend": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        "m2Bend": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      }
+  }
+
+  # convert dictionary to yaml string
+  config = yaml.safe_dump(ofc_config)
+
+  await mtaos.cmd_runOFC.set_start(wf = wf, config = config)
+
+Similarly, the user can manipulate the configuration for Camera hexapod, M2 and M1M3.
+Furthermore, this gives users the capability of customizing the configuration of the OFC on the fly, without the need to recycle the CSC state and the need for several different configurations.
+By relying on a yaml string to parse the configuration it is also possible to expand the feature as needed without the need to update the xml interface, which considerably simplify the redeployment cycle.
+
+.. _Architecture-updates:
+
+Architecture
+============
+
+In terms of the MTAOS itself, there is practically no substantial architecture changes needed (see :numref:`fig-mtaosClass` bellow).
+The current code-base is well organized in the sense that the "heavy lifting" is encapsulated by the `Model class`_.
+
+.. figure:: /_static/mtaosClass.png
+   :name: fig-mtaosClass
+   :target: ../_images/mtaosClass.png
+   :alt: MTAOS class diagram
+
+   MTAOS class diagram (`original uml file from v0.4.5`_).
+
+
+.. _Model class: https://ts-mtaos.lsst.io/py-api/lsst.ts.MTAOS.Model.html#lsst.ts.MTAOS.Model
+.. _original uml file from v0.4.5: https://github.com/lsst-ts/ts_MTAOS/blob/v0.4.5/doc/uml/mtaosClass.uml
+
+Nevertheless, the `Model class`_ itself will require substantial updates to incorporate the new functionality mentioned :ref:`above <Data-access-changes>` and the conversion of `WEP`_ into a gen3 pipeline task.
+
+Updates to the Model class
+--------------------------
+
+The `Model class`_ has two main attributes it uses to perform the wavefront estimation (`WEP`_) and to convert the output to optical corrections (`OFC`_).
+
+The ``wep`` attribute (and its uses) is probably the one that will require the most work.
+First, we have to keep in mind that the ``wep`` will have two modes of operation; local and remote.
+
+The local mode will be used for the closed loop, e.g., analysis of the corner wavefront sensors for the main camera.
+Whereas, remote mode will be used (mostly) for full focal plane analysis (specially for ``LSSTCam``).
```

Comment:
I think in this case, the DM team can use the proposed `ts_wep` pipeline task to calculate the wavefront error and publish the result directly instead of using the `Model` class in `ts_MTAOS`. Actually, I do not think it is a good idea for DM team to use `ts_MTAOS` directly in `LsstCam`.